src/reanimated2/SetNativeProps.ts

```diff
@@ -26,6 +26,11 @@ if (isWeb()) {
 } else if (IS_NATIVE && global._IS_FABRIC) {
   setNativeProps = (animatedRef, updates) => {
     'worklet';
+    if (!_WORKLET) {
+      throw new Error(
+        '[Reanimated] `setNativeProps` can only be used on the UI runtime. Please call the function with `runOnUI` or run it in a worklet instead.'
```

Comment:
While this comment is definitely good, I can't help but think that "run it in a worklet instead" could be misleading at times. Some worklets are called on JS thread, even deliberately. Also we have such cases like:

```TS
function MyInnerComponent(props: { someRef: any }) {
  const animatedStyle = useAnimatedStyle(() => {
    setNativeProps(props.someRef, {});
    return {};
  });
  return <Animated.View style={animatedStyle} />;
}

function MyOuterComponent() {
  const someRef = React.useRef(null);
  return (
    <View ref={someRef}>
      <MyInnerComponent someRef={someRef} />
    </View>
  );
}
```

`useAnimatedStyle` executes on JS thread first, therefore we get crash here (I checked). User would have to check first if he's on UI thread first, same as it was with `measure` and we don't want that.

I would lean more toward the way we handle it with `measure` - silently failing - but I'm open for discussion.

Reply:
I agree with Tomek that "run it in a worklet instead" can be misleading in this case. `setNativeProps` is meant only to be used on the UI runtime. In this sense, it's kinda similar to `measure`. However, a silent fail is not a great solution here because `setNativeProps` has no return value and `measure` simply returns `null` (which is also sometimes does on the UI thread so the developer needs to handle it). I'd suggest using `console.warn` here with an error message "[Reanimated] `setNativeProps` can only be used on the UI runtime." which effectively is a no-op in release mode.

Reply:
But then in this case you'd get constant warnings in the console. And we all know the tragic story from gesture handler about this...

Reply:
Yep but only if you use it incorrectly (i.e. from the JS thread) and also it's a totally new feature so there should be no regressions.

Reply:
It won't be no-op in case of a warn because without this error it crashes later on with:

![image](https://github.com/software-mansion/react-native-reanimated/assets/39658211/7efbf521-f408-4f9f-9c29-a4ac6610e416)

because `animatedRef.viewName.value` is undefined on JS runtime

Reply:
I've updated the error message to "[Reanimated] `setNativeProps` can only be used on the UI runtime." in [20be9ea](https://github.com/software-mansion/react-native-reanimated/pull/5043/commits/20be9eae511b783a1914c1a91044cd1b0de1d68d)

Reply:
I made it no-op in the end 😅 

![image](https://github.com/software-mansion/react-native-reanimated/assets/39658211/6d7c1f51-8304-4f61-b918-54ef1afccca5)